internal/earlyconfig/config.go

```diff
@@ -170,7 +170,7 @@ func (c *Config) ProviderDependencies() (*moduledeps.Module, tfdiags.Diagnostics
 			fqn = addr
 		}
 		if fqn.IsZero() {
-			fqn = addrs.NewLegacyProvider(name)
+			fqn = addrs.NewDefaultProvider(name)
```

Comment:
We might as well default to "NewDefaultProvider" instead of "NewLegacyProvider" here; it's an error in either case - we *shouldn't* be running into empty FQNs at this point.